### PR TITLE
chore(reference): remove legacy brower caution tips 

### PR DESCRIPTION
**Pull request by rwaskiewicz**
this commit removes all caution and warning tips for legacy browsers. in
some cases, the surrounding text needed copy edits (as it referenced
supporting 1+ legacy browser(s). the text has been modified/outright
removed where appropriate.

this commit updates the browser support page to remove references to
legacy browsers. it also removes mention of a versioning matrix that no
longer applies (sort of).

### CODE REVIEW

File: docs/reference/browser-support.md [hunk 1]
```diff
@@ -7,7 +7,10 @@ slug: /browser-support
 
 # Browser Support
 
-Stencil builds Web Components that run natively or near-natively in all widely used desktop and mobile browsers.
+Stencil builds Web Components that run natively in all widely used desktop and mobile browsers.
```
Comment (rwaskiewicz) [inline docs/reference/browser-support.md:10]: This page looks something like this now:
Screenshot 2023-06-14 at 3 55 26 PM

I wonder if it makes sense to roll this into our support policy (https://stenciljs.com/docs/support-policy) and just setup a redirect. The page is kinda bare and doesn't really warrant its own route anymore (IMO). Thoughts?

File: src/css/bs-chart.scss [hunk 2]
```diff
@@ -14,23 +14,21 @@
...
   border-radius: 8px;
   color: #fff;
   padding: 8px;
   font-size: 11px;
   font-weight: 600;
+
+  &--deprecated {
+    @extend .bs-chart__card;
+    background: #d0901a;
+  }
```
Comment (rwaskiewicz) [inline src/css/bs-chart.scss:31]: Styles don't get copied/propagated across versions. So when I removed the IE 11 bubble from the `docs/browser-support` page, that colored other bubble this fun mustard yellow instead.

If we end up rolling this (`docs/browser-support.md`) page into the support page, I'll roll back this change (and the changes to versioned browser-support pages)

File: docs/reference/browser-support.md (hunk 1 above)
Comment (alicewriteswrongs) [reply docs/reference/browser-support.md:10]: it's kind of a stub now, I'm down with merging 👍

File: docs/framework-integration/angular.md [hunk 3]
```diff
@@ -516,19 +516,6 @@ between frameworks.
 You can configure how your input events can map directly to a value accessor, allowing two-way data-binding to be a built in feature of any of your
 components. Take a look at [valueAccessorConfig's option above](#valueaccessorconfigs).
 
-### How do I add IE11 or Edge support?
```
Comment (alicewriteswrongs) [inline docs/framework-integration/angular.md:519]: you don't! haha

**Review by alicewriteswrongs — APPROVED** [review]
lgtm!

File: docs/reference/browser-support.md (hunk 1 above)
Comment (rwaskiewicz) [reply docs/reference/browser-support.md:10]: Yeah, I looked at this this morning with a fresh set of eyes - I think merging makes the most sense to me

File: docs/reference/browser-support.md
Comment (rwaskiewicz) [inline docs/reference/browser-support.md:1]: Merged contents into docs/reference/support-policy.md

File: docs/reference/support-policy.md [hunk 4]
```diff
@@ -59,6 +59,19 @@ In the example above, when Version D is released, Version C enters maintenance m
...
+Stencil builds Web Components that run natively in all widely used desktop and mobile browsers.
+Custom Elements are natively supported in Chrome, Edge, Firefox, and Safari (including iOS)!
+
+Stencil supports the following browsers:
+
+| Stencil Version | Chrome | Edge | Firefox | Safari | Internet Explorer | Pre-Chromium Edge |
+|:---------------:|:------:|:----:|:-------:|:------:|:-----------------:|:-----------------:|
+|       V4        |  v79+  | v79+ |  v70+   |  v14+  |     &#10060;      |     &#10060;      |
+|       V3        |  v79+  | v79+ |  v70+   |  v14+  |      &#9989;      |      &#9989;      |
+|       V2        |  v60+  | v79+ |  v63+   | v10.1+ |      &#9989;      |      &#9989;      |
```
Comment (rwaskiewicz) [inline docs/reference/support-policy.md:73]: Unsure what the exact v1 policy was (I wrote the browser support policy). The wayback machine wasn't much help (there wasn't page to crawl back to), and the Stencil `BREAKING_CHANGES.md` doc in the Core repo wasn't helpful either. If we feel we _need_ a v1 row for completeness, we can always just copy the v1 row - it's not like v1 is maintained anymore

File: vercel.json [hunk 5]
```diff
@@ -145,6 +145,11 @@
       "source": "/:version(next|(?:v[\\d]+(?:\\.[\\dx]+)?))?/stencil-redux",
       "destination": "https://github.com/ionic-team/stencil-redux",
       "statusCode": 301
+    },
+    {
+      "source": "/:version(next|(?:v[4-9]+(?:\\.[\\dx]+)?))?/browser-support",
```
Comment (rwaskiewicz) [inline vercel.json:150]: I'm _pretty sure_ this'll work.  Unfortunately we can't test these w/o deploying :-(

**Comment by rwaskiewicz** [comment]
@alicewriteswrongs I combined the two pages - would you mind taking a second pass? This PR now has 100% less SCSS changes 🎉

**Review by alicewriteswrongs — APPROVED** [review]
all looks good!

File: docs/reference/support-policy.md (hunk 4 above)
Comment (sean-perkins) [reply docs/reference/support-policy.md:73]: I think it's fine to leave v1 off. If it gets requested we can make some best guesses.

**Review by sean-perkins — APPROVED** [review]

**Pull request merged**